### PR TITLE
nbsphinx: add Sphinx plugin to show Jupyter notebooks

### PR DESCRIPTION
**Pull request by jia200x**
This PR:
- Adds the `nbsphinx` plugin to include Jupyter notebooks in Sphinx documentation
- Move the `index.rst` and `conf.py` to the root directory (unfortunately we will need to do this until someone finds a cleaner approach to import opensir outside of the root directory)
- Fix some style and import issues in the Jupyter notebook
- Include the notebook from Sphinx
- Add `pandoc` to Dockerfile (required by nbsphinx)

This indirectly adds some tests for Jupyter notebooks. If there's something wrong (style issues or wrong python code) then the documentation won't build (and CircleCI will complain)

Test with:
```
pipenv run doc
firefox doc/build/html
```

### CODE REVIEW

**Comment by jia200x** [comment]
I found the problem. It seems to be PEBKAC

**Comment by felipehuerta17** [comment]
To put the docs back in the doc folder, have you tried adding the following lines to conf.py?

```python
path_opensir = '../'
sys.path.append(path_opensir)
```

File: Pipfile [hunk 1]
```diff
@@ -17,6 +17,7 @@ sklearn = "*"
 sphinx = "*"
 sphinx-rtd-theme = "*"
 toml = "*"
+nbsphinx = "*"
```
Comment (sasalatart) [inline Pipfile:20]: I see we have many packages that maybe should be in the `dev-packages` section? I am not sure though. What do you think? I am not 100% sure if these documentation packages should be dev-dependencies or not.

Comment (jia200x) [reply Pipfile:20]: For now I think it should be ok to move them to `dev-packages`.
At some point, we might be able to generate `dev` and `prod` documentation, so we would need both packages there.

How to proceed with this?

Comment (jia200x) [reply Pipfile:20]: @felipehuerta17 @leandrolanzieri any opinions on this?

**Comment by jia200x** [comment]
> To put the docs files back in the doc folder, have you tried adding the following lines to conf.py?

I tried it, but for some reason Sphinx is not able to find the files :(.
If someone is able to do it it would be cool

**Comment by jia200x** [comment]
I have no idea why Circle CI fails to generate the documentation. I cannot reproduce it locally with the Dockerfile

**Comment by leandrolanzieri** [comment]
I get the same error as the CI locally. Will investigate

File: Pipfile (hunk 1 above)
Comment (felipehuerta17) [reply Pipfile:20]: I am not familiar what's good practice in this domain. For the moment, I don't have a problem on having a long list, as the user can still do `git clone, cd open-sir and pip install.`. And I think that completeness and accuracy is priority over conciseness, at this stage.

**Comment by felipehuerta17** [comment]
Hi @jia200x , I am getting the following error:
image
Are you getting the same?

**Comment by jia200x** [comment]
> Hi @jia200x , I am getting the following error:
image
Are you getting the same?

I added a "DELETE_ME" comment to print whatever is happening in the CI. That's why you see that error. I will remove the commit in a few minutes

**Comment by jia200x** [comment]
FINALLY!!! fixed it adding a pandoc dependency to the build process

**Comment by jia200x** [comment]
I updated the README to include the "pandoc" dependency when building the documentation.

**Comment by jia200x** [comment]
The error is expected. I rebased, and there are some errors in the Jupyter. I'm fixing them

**Comment by jia200x** [comment]
@felipehuerta17 @sasalatart @leandrolanzieri OK, after several cups of coffee all tests pass. Mind to give it another look?

File: Pipfile (hunk 1 above)
Comment (jia200x) [reply Pipfile:21]: ok, I propose this: let's proceed with this one and I open an issue to decide the boundaries between pipenv (dev, prod) and setuptools (pip install opensir).

Does this sound reasonable?

**Comment by leandrolanzieri** [comment]
LGTM, Go!

**Pull request merged**